pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,57 @@
+package v1
+
+import (
+	"net"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//MultiClusterService describes the information of service that is available across
+//clusters. A shared informer running in the lighthouse control plane will listen for
+//service creation in every cluster that is part of the control plane and whenever a
+//service is created a MultiClusterService will be created. Lighthouse will
+//distribute the MultiClusterService to every other cluster that is the part of the
+//control plane. When the service is updated/deleted lighthouse ensures that the
+//updated state is reflected in every cluster.
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MultiClusterServiceSpec `json:"spec"`
+}
+
+//MultiClusterServiceSpec provides the specification of a MultiClusterService
+type MultiClusterServiceSpec struct {
+	//List of service information across clusters. This field is not optional. Each
+	//service name should have at least one service info associated with it.
+	Items []ClusterServiceInfo `json:"items"`
+}
+
+//ClusterServiceInfo provides information about service in each cluster.
+type ClusterServiceInfo struct {
+	//The unique identity of the cluster where the service is available. This field is
+	//not optional.
+	ClusterID string `json:"clusterId"`
```

Comment:
Not trying to nit but the recommended convention for naming "ids" is uppercase, ie `clusterID` ...